Update SQL spec descriptions

jaredmoo opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [x] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

jaredmoo committed:
More documentation of editon, pool name, and slo.

jaredmoo committed:
"Azure SQL Database" -> "database", etc

Reading "Azure SQL" over and over again is exhausting and gets in the way of understanding the docs.

azurecla commented:
Hi __@jaredmoo__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (jaredmoo). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

nathannfan commented on arm-sql/2014-04-01/swagger/firewallRules.json:
```diff
@@ -241,7 +241,7 @@
             "$ref": "#/definitions/ServerFirewallRule"
           },
-          "description": "The list of Azure SQL server firewall rules for the server."
+          "description": "The list of server firewall rules for the server."
```
Can you just make it "The list of firewall rules for the server" for this one?

nathannfan replied:
Can you just make it "The list of firewall rules for the server" for this one?

nathannfan commented on arm-sql/2014-04-01/swagger/sql.core.json:
```diff
@@ -2,7 +2,7 @@
   "swagger": "2.0",
   "info": {
     "title": "Azure SQL Database API spec",
-    "description": "Provides create, read, update and delete functionality for Azure SQL resources including servers, databases, elastic pools, recommendations, operations, and usage metrics.",
```
This is the one place I'd like to leave Azure SQL

nathannfan replied:
This is the one place I'd like to leave Azure SQL in

nathannfan commented on arm-sql/2014-04-01/swagger/sql.core.json:
```diff
@@ -294,7 +294,7 @@
         ],
         "operationId": "Databases_ListByServer",
-        "description": "Returns information about an Azure SQL database.",
+        "description": "Returns information about a database.",
```
I know this isn't your error, but could you put "Returns a list of databases by server."?

nathannfan commented on arm-sql/2014-04-01/swagger/sql.core.json:
```diff
@@ -375,7 +375,7 @@
         ],
         "operationId": "Servers_List",
-        "description": "Returns information about an Azure SQL server.",
+        "description": "Returns information about a server.",
```
"Returns a list of servers."?

nathannfan commented on arm-sql/2014-04-01/swagger/sql.core.json:
```diff
@@ -554,7 +554,7 @@
         ],
         "operationId": "Servers_ListUsages",
-        "description": "Returns information about Azure SQL server usage.",
+        "description": "Returns information about server usage.",
```
usages

nathannfan commented on arm-sql/2014-04-01/swagger/sql.core.json:
```diff
@@ -943,7 +943,7 @@
         ],
         "operationId": "RecommendedElasticPools_Get",
-        "description": "Gets information about an Azure SQL Recommended Elastic Pool.",
+        "description": "Gets information about an recommended elastic pool.",
```
a recommended

jaredmoo committed:
"server firewall rule" -> "firewall rule".

Also restored "Azure SQL Database" wording to doc titles and tweaked list operation descriptions.

nathannfan commented on arm-sql/2014-04-01/swagger/sql.core.json:
```diff
@@ -1560,7 +1560,7 @@
         }
       ],
-      "description": "Represents an Azure SQL Recommended Elastic Pool."
+      "description": "Represents an recommended elastic pool."
```
Maybe ctrl+replace 'an recommended'

nathannfan replied:
Maybe ctrl+replace 'an recommended'

jaredmoo committed:
Fixed "an recommended"

anuchandy reviewed (commented):
CI is failing specifically with error "FATAL: Error generating client model: System.InvalidOperationException: Swagger document contains two or more x-ms-enum extensions with the same name 'ServiceObjectiveName' and different values.", please take a look and fix.

jaredmoo committed:
Fixed firewall rule description type

jaredmoo committed:
Fixed inconsistent ServiceObjectiveName enum.

jaredmoo commented:
Fixed, thanks. Node build seems to be hitting a transient error.

anuchandy commented:
@jaredmoo thanks, restarted job and now it is passing.

jaredmoo commented:
Can I get a review and merge here?

jaredmoo committed:
Additional changes to op descriptions
- Removed 'information about' wording
- Fixed 'an data warehouse'
- Made tense consistent
- Corrected TDE description

anuchandy commented:
@jaredmoo thanks, merging.

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-ruby.svg?branch=RestAPI-PR970)](https://travis-ci.org/AutorestCI/azure-sdk-for-ruby) https://github.com/Azure/azure-sdk-for-ruby/pull/700

AutorestCI commented:
No modification for NodeJS

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-python.svg?branch=RestAPI-PR970)](https://travis-ci.org/AutorestCI/azure-sdk-for-python) https://github.com/Azure/azure-sdk-for-python/pull/1010

jaredmoo commented:
Thanks!
